### DISCUSSION TITLE
Regression on shared with 1.16.1

### DISCUSSION THREAD

**Discussion by Alex293** [pointfreeco/swift-composable-architecture#3507] [General]
### Description

Hi, we noticed a regression in our tests, we identified that the regression was introduced in 1.16.1 ; 1.16.0 is working as expected. I'll update the issue with a failing test case / reproducer ASAP. The issue is 100% deterministically reproducible.

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

Shared's value should be readable synchronously right after writing to it.
ex :
in state : 
```swift
@Shared // in memory shared, no persistence key
var someShared: Shared<SomeValueType?>
```
In reducer :
```swift
case .action: 
    state.someShared = someValue
    return .none // break point here
```
expected: 
`po state.someShared != nil` should be `true`

### Actual behavior

Same code, no change between test except the TCA version, clean
Shared's value is nil.
`po state.someShared != nil` yields `false`

### The Composable Architecture version information

1.16.1

### Destination operating system

18.0, 18.2, 17.0 (initial list of tested versions, I expect this is not iOS version related)

### Xcode version information

Xcode 16.0, 16.1, 16.2 (beta)

### Swift Compiler version information

```shell
swift-driver version: 1.115.1 Apple Swift version 6.0.3 (swiftlang-6.0.3.1.4 clang-1600.0.30)
Target: arm64-apple-macosx15.0

Reproducible too with toolchains from the mentioned Xcode versions
```

**Comment by mbrandonw** [comment]
Hi @Alex293, can you please provide a test that fails that demonstrates the problem you are seeing? I can't reproduce this:

<img width="404" alt="image" src="https://github.com/user-attachments/assets/f8d142b5-bd8b-4f6f-9739-9e4c6f7fc5c7">

Also, nothing stands out to me in the [diff between 1.16 and 1.16.1](https://github.com/pointfreeco/swift-composable-architecture/compare/1.16.0...1.16.1) to explain this. For the moment I'm not sure there is a regression here and so I am going to convert this to a discussion. If you can provide a failing test we can convert this back to an issue.

**Reply by Alex293** [reply]
Yes I'll provide one soon, I posted a short extract in case someone stumble on the same issue, reproducing requires more setup.

**Reply by Alex293** [reply]
Ok sorry for the alarming bug report, this was due to a faulty codable implementation (no round trip). As we don't use any Shared storage I still can't explain why it's working with previous versions though
